safe/gui/tools/print_report_dialog.py

```diff
@@ -0,0 +1,486 @@
+    get_ui_class, resources_path, html_header, html_footer)
+
+__copyright__ = "Copyright 2016, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+INFO_STYLE = styles.BLUE_LEVEL_4_STYLE
+FORM_CLASS = get_ui_class('print_report_dialog.ui')
+LOGGER = logging.getLogger('InaSAFE')
+
+
+class PrintReportDialog(QtGui.QDialog, FORM_CLASS):
+    """Print report dialog for the InaSAFE plugin."""
+
+    def __init__(self, impact_function, iface, dock=None, parent=None):
+        """Constructor for the dialog.
+        :param iface: A Quantum GIS QGisAppInterface instance.
+        :type iface: QGisAppInterface
+
+        :param parent: Parent widget of this dialog
+        :type parent: QWidget
+
+        :param dock: Optional dock widget instance that we can notify of
+            changes to the keywords.
+        :type dock: Dock
+
+        .. versionadded: 4.3.0
+        """
+
+        QtGui.QDialog.__init__(self, parent)
+        self.setupUi(self)
+
+        # additional buttons
+        self.button_print_pdf = QtGui.QPushButton(self.tr('Print as PDF'))
+        self.button_print_pdf.setObjectName('button_print_pdf')
+        self.button_print_pdf.setToolTip(self.tr(
+            'Write report to PDF and open it in default viewer'))
+        self.button_box.addButton(
+            self.button_print_pdf, QtGui.QDialogButtonBox.ActionRole)
+
+        self.button_open_composer = QtGui.QPushButton(
+            self.tr('Open in composer'))
+        self.button_open_composer.setObjectName('button_open_composer')
+        self.button_open_composer.setToolTip(
+            self.tr('Prepare report and open it in QGIS composer'))
+        self.button_box.addButton(
+            self.button_open_composer, QtGui.QDialogButtonBox.ActionRole)
+
+        self.button_print_pdf.clicked.connect(self.accept)
+        self.button_open_composer.clicked.connect(self.accept)
+
+        # Save reference to the QGIS interface and parent
+        self.iface = iface
+        self.parent = parent
+        self.dock = dock
+        self.impact_function = impact_function
+
+        self.create_pdf = False
+
+        self.default_template_radio.toggled.connect(
+            self.toggle_template_selectors)
+
+        # Set up things for context help
+        self.help_button = self.button_box.button(QtGui.QDialogButtonBox.Help)
+        # Allow toggling the help button
+        self.help_button.setCheckable(True)
+        self.help_button.toggled.connect(self.help_toggled)
+        self.main_stacked_widget.setCurrentIndex(1)
+
+        self.unwanted_templates = ['merged_report.qpt', 'infographic.qpt']
+
+        # Load templates from resources...
+        template_dir_path = resources_path('qgis-composer-templates')
+        self.populate_template_combobox(
+            template_dir_path, self.unwanted_templates)
+
+        #  ...and user directory
+        settings = QtCore.QSettings()
```

Comment:
We are remove the use of QSettings in the code. Can you use safe/utilities/settings.py, thanks. Same for use of QSettings in this PR

Reply:
+1